atlauncher: 3.4.30.0 -> 3.4.34.0

GetPsyched opened the pull request:
## Description of changes

1. Update the package
2. Fix icon that didn't show up

Changelog available [here](https://atlauncher.com/downloads)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

GetPsyched committed:
atlauncher: 3.4.30.0 -> 3.4.34.0

GetPsyched committed:
atlauncher: fix icon

pbsds commented on pkgs/games/atlauncher/default.nix:
```diff
@@ -2,11 +2,16 @@
+  };
+
+  icon = fetchurl {
+    url = "https://avatars.githubusercontent.com/u/7068667";
```
This URL is not really guaranteed to be stable. The file could change and invalidate the hash. Do they host it on any repo? Since it is in the nixpkgs cache it wont likely break, but still.

GetPsyched replied:
Hmm, I'll try to find a stable URL for this.

pbsds commented on pkgs/games/atlauncher/default.nix:
```diff
@@ -2,11 +2,16 @@
+    hash = "sha256-gHUYZaxADchikoCmAfqFjVbMYhhiwg2BZKctmww1Mlw=";
+  };
+
+  icon = fetchurl {
```
```suggestion
  env.ICON = fetchurl {
```

pbsds commented on pkgs/games/atlauncher/default.nix:
```diff
@@ -30,17 +35,17 @@ stdenv.mkDerivation (finalAttrs: {
           }
+
+      mkdir -p $out/share/icons/hicolor/128x128/apps
+      cp ${finalAttrs.icon} $out/share/icons/hicolor/128x128/apps/${finalAttrs.pname}.png
```
```suggestion
      cp $ICON $out/share/icons/hicolor/128x128/apps/${finalAttrs.pname}.png
```

Makes it overridable

GetPsyched committed:
atlauncher: refactor install phase

- move working dir from `.atlauncher` to `.local/share/ATLauncher`
- remove steam-run as a dependency
- fix no-launcher-update not working when launching from CLI

GetPsyched commented:
Hey, @pbsds, thanks for the review! I've made the changes you requested and added a refactor of the install phase as well. I've mentioned the rational in the commit description.

GetPsyched commented:
Weird, how did ofborg detect a new package when there is none.

pbsds commented on pkgs/games/atlauncher/default.nix:
```diff
@@ -1,46 +1,47 @@
+    makeWrapper ${jre}/bin/java $out/bin/atlauncher \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ xorg.libXxf86vm udev ]}" \
+      --add-flags "-jar $out/share/java/ATLauncher.jar" \
+      --add-flags "--working-dir \''${XDG_DATA_HOME:-\$HOME/.local/share}/ATLauncher" \
```
```suggestion
      --add-flags "--working-dir \"\''${XDG_DATA_HOME:-\$HOME/.local/share}/ATLauncher\"" \
```

pbsds commented on pkgs/games/atlauncher/default.nix:
```diff
@@ -1,46 +1,47 @@
+    mkdir -p $out/bin $out/share/java
+    cp $src $out/share/java/ATLauncher.jar
+
+    makeWrapper ${jre}/bin/java $out/bin/atlauncher \
```
```suggestion
    makeWrapper ${jre}/bin/java $out/bin/${finalAttrs.pname} \
```

This line must match the `exec` in `desktopItems`

pbsds reviewed (changes requested):
Result of `nixpkgs-review pr 257299` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>atlauncher</li>
  </ul>
</details>

---

- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] all depending packages build
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

---

Just a couple of nits

GetPsyched commented:
Hey, @pbsds, thanks for the review!

One more thing, this PR adds a breaking change (moving the working directory from `~/.atlauncher` to `~/.local/share/ATLauncher`). This would mean anyone using this package would need to move their existing data to the newer directory.

Is there a process to mark this through `nixpkgs` or?

GetPsyched commented:
Also, a very optional thing but, how can I make the initial window that shows up to float? I use a tiling WM so that initial window gets stretched into oblivion. It doesn't matter much because it's only shown for 1-2 seconds after which the actual window shows.

Worth noting that it also matters for the console window feature of ATLauncher that also should be floating.

pbsds commented:
> One more thing, this PR adds a breaking change (moving the working directory from `~/.atlauncher` to `~/.local/share/ATLauncher`).

If this change is done upstream and not by you then it's on them. Consider submitting an issue. You could be nice and add a wrapper script which checks if `~/.atlauncher` exists, and then symlink it. This change however i would say means you cannot backport it to 23.05.

> ... how can I make the initial window that shows up to float? ...

Also a upstream problem. Perhaps you can tag the process using xprop or whatever is the equivalent in wayland.
